src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1632,41 +1648,39 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
         if (_searcher.ResetIfStale(*GetRenderData(), text, !goForward, !caseSensitive))
         {
-            _searcher.HighlightResults();
-            _searcher.MoveToCurrentSelection();
             _cachedSearchResultRows = {};
+            if (SnapSearchResultToSelection())
+            {
+                _searcher.MoveToCurrentSelection();
+                SnapSearchResultToSelection(false);
+            }
         }
         else
         {
             _searcher.FindNext();
         }
 
-        const auto foundMatch = _searcher.SelectCurrent();
-        auto foundResults = winrt::make_self<implementation::FoundResultsArgs>(foundMatch);
-        if (foundMatch)
+        auto evArgs = winrt::make_self<implementation::UpdateSearchResultsEventArgs>();
+        evArgs->State(SearchState::InActive);
+        if (!text.empty())
         {
-            // this is used for search,
-            // DO NOT call _updateSelectionUI() here.
-            // We don't want to show the markers so manually tell it to clear it.
-            _terminal->SetBlockSelection(false);
-            _UpdateSelectionMarkersHandlers(*this, winrt::make<implementation::UpdateSelectionMarkersEventArgs>(true));
-
-            foundResults->TotalMatches(gsl::narrow<int32_t>(_searcher.Results().size()));
-            foundResults->CurrentMatch(gsl::narrow<int32_t>(_searcher.CurrentMatch()));
-
-            _terminal->AlwaysNotifyOnBufferRotation(true);

```

Comment:
little sus that this is missing now... note to self, make sure that search marks still update when search box is open? ...

Reply:
I answered this to myself: it's down in `TermControl::_coreUpdateSearchResults`